core/src/test/kotlin/org/jetbrains/jewel/PainterHintTest.kt

```diff
@@ -0,0 +1,122 @@
+package org.jetbrains.jewel
+
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.state.ToggleableState
+import org.jetbrains.jewel.painter.PainterHint
+import org.jetbrains.jewel.painter.PainterPathHint
+import org.jetbrains.jewel.painter.PainterResourcePathHint
+import org.jetbrains.jewel.painter.hints.Dark
+import org.jetbrains.jewel.painter.hints.Override
+import org.jetbrains.jewel.painter.hints.Selected
+import org.jetbrains.jewel.painter.hints.Size
+import org.jetbrains.jewel.painter.hints.Stateful
+import org.jetbrains.jewel.painter.hints.Stroke
+import org.jetbrains.jewel.painter.rememberResourcePainterProvider
+import org.junit.Assert
+import org.junit.Test
+
+@Suppress("ImplicitUnitReturnType")
+class PainterHintTest : BasicJewelUiTest() {
+
+    @Test
+    fun `empty hint should be ignored`() = runComposeTest({
+        CompositionLocalProvider(LocalIsDarkTheme provides false) {
+            val provider = rememberResourcePainterProvider("icons/github.svg", PainterHintTest::class.java)
+
+            val painter1 by provider.getPainter()
+            // must be ignored the None and hit cache
+            val painter2 by provider.getPainter(PainterHint.None)
+            // must be ignored the None and hit cache too
+            val painter3 by provider.getPainter(PainterHint.None, PainterHint.None)
+
+            Assert.assertEquals(painter1, painter2)
+            Assert.assertEquals(painter3, painter2)
+        }
+    }) {
+        awaitIdle()
+    }
+
+    private fun String.applyPathHints(vararg hints: PainterHint): String {
+        var result = this
+        hints.forEach {
+            result = when (it) {
+                is PainterResourcePathHint -> it.patch(result, emptyList())
+                is PainterPathHint -> it.patch(result)
+                else -> return@forEach
+            }
+        }
+        return result
+    }
+
+    @Test
+    fun `dark painter hint`() {
+        val basePath = "icons/github.svg"
+        basePath.applyPathHints(Dark(true)).let {
+            Assert.assertEquals("icons/github_dark.svg", it)
+        }
+    }
```

Comment:
```suggestion
    @Test
    fun `dark painter hint should append suffix when isDark is true`() {
        val basePath = "icons/github.svg"
        val patchedPath = basePath.applyPathHints(Dark(true))
        Assert.assertEquals("icons/github_dark.svg", patchedPath)
    }

    @Test
    fun `dark painter hint should not append suffix when isDark is false`() {
        val basePath = "icons/github.svg"
        val patchedPath = basePath.applyPathHints(Dark(false))
        Assert.assertEquals(basePath, patchedPath)
    }
```

I would test all possible cases, here and for the other hints. For hints with a single boolean parameter that means checking both true and false cases. And in general, I'd try to have only one assertion per test case, because that makes it easier to figure out what gets broken :)